src/osd/PGLog.h

```diff
@@ -335,6 +335,30 @@ struct PGLog : DoutPrefixProvider {
       unsigned split_bits,
       IndexedLog *target);
 
+    void split_pwlc(pg_info_t &info) {
+      eversion_t previous_version;
+      if (rollback_info_trimmed_to_riter == log.rend()) {
+	previous_version = tail;
+      } else {
+	previous_version = rollback_info_trimmed_to_riter->version;
+      }
+      // When a split occurs log entries are divided between the two PGs,
+      // this can leave pwlc refering to entries that are no longer in this
+      // PG log. Update pwlc so it is not beyond the last entry in the log.
+      // Non-primary shards which don't have a full log may rollback pwlc
+      // too far, but this will get corrected by the primary shard when
+      // activating shards later in peering.
+      for (auto & [shard, versionrange] : info.partial_writes_last_complete) {
```

Comment:
Hmm, for-FastEC pools all that has been done is just setting `previous_version` local variable. Seems fine.